### PR TITLE
Change issue types and statuses to strings

### PR DESCRIPTION
**Pull request by michpetrov**
Issue: [SET-245](https://projects.engineering.redhat.com/browse/SET-245)

### CODE REVIEW

**Comment by rstancel** [comment]
@soul2zimate Chao, can you please take a look? thanks

File: jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    public static final String RETEST = "Retest";
+    public static final String START_PROGRESS = "Start Progress";
+    public static final String START_QA = "Start QA";
+    public static final String STOP_PROGRESS = "Stop Progress";
+    public static final String STOP_QA = "Stop QA";
+    public static final String UPDATE_PULL_REQUEST = "Update Pull Request";
+    public static final String VERIFY_ISSUE = "Verify Issue";
+
+
+    static final Map<String, Map<String, String>> TRANSITION_TABLE = new ImmutableMap.Builder<String, Map<String, String>>()
```
Comment (soul2zimate) [inline jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java:30]: Does this transition table only bind to JBEAP issue workflow ? cc @baranowb

File: domain/src/main/java/org/jboss/set/aphrodite/domain/IssueType.java [hunk 2]
```diff
@@ -22,25 +22,7 @@
...
-
-    public static IssueType getMatchingIssueType(String value) {
-        for (IssueType type : IssueType.values())
-            if (type.get().equalsIgnoreCase(value))
-                return type;
-
-        return UNDEFINED;
-    }
+public class IssueType {
+    public static final String UNDEFINED = "UNDEFINED";
```
Comment (soul2zimate) [inline domain/src/main/java/org/jboss/set/aphrodite/domain/IssueType.java:26]: If you have no need to shared enum among other classes, you can remove this class and use these two directly.

File: jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package org.jboss.set.aphrodite.issue.trackers.jira;
```
Comment (soul2zimate) [inline jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java:1]: need a template class header

**Comment by baranowb** [comment]
What is rationale behind this change?

File: jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java (hunk 1 above)
Comment (michpetrov) [reply jira/src/main/java/org/jboss/set/aphrodite/issue/trackers/jira/JiraTransitions.java:30]: Yes, this is for the workflow used by JBEAP, it's a refactor/update of the current code (currently not all transitions are covered). I'm not entirely sure about the purpose of this but changing it entirely is outside the scope of this PR.

**Comment by michpetrov** [comment]
@baranowb as mentioned in the Jira the types/statuses coming out of Aphrodite are different from what we see in Jira which is atm rather counter-productive.

**Comment by baranowb** [comment]
Had to look through a bit. AFAIR its not modeled after BZ. Local representation of issue, be it JIRA or BZ was meant to unify handling regardless of tracker, hence mapping to common values. Switching to string will make devs lookup tracker constantly for valid values. If there is some bug as to "lost values" it has to be fixed in mapping.

**Comment by michpetrov** [comment]
@baranowb Well, any suggestions then? The issue here is that we need to know the actual Jira values (to which end the "common values" are counter-productive).

The way I'm looking at it the only other good solution is to keep a reference to the original issue but that makes you dependent on the APIs Aphrodite is shielding you from. Unless we add a bunch of getters like `getOriginalStatus()` etc.

**Comment by michpetrov** [comment]
Since we're scrapping the Strings idea I'll close this one and send another.

**Pull request closed without merging**